### ISSUE TITLE
eCAL@Windows: cannot cmake against ecal due to not found eCAL::mon_plugin_lib

### ISSUE THREAD

**Issue by FlorianReimold** [eclipse-ecal/ecal#789]
**Describe the bug**
Currently it is not possible to link against the eCAL Master, as CMake is missing the mon_plugin_lib target:

``` txt
CMake Error at C:/eCAL/lib/cmake/eCAL/eCALTargets-release.cmake:89 (set_property):
  set_property could not find TARGET eCAL::mon_plugin_lib.  Perhaps it has
  not yet been created.
Call Stack (most recent call first):
  C:/eCAL/lib/cmake/eCAL/eCALTargets.cmake:161 (include)
  C:/eCAL/lib/cmake/eCAL/eCALConfig.cmake:62 (include)
  CMakeLists.txt:9 (find_package)


CMake Error at C:/eCAL/lib/cmake/eCAL/eCALTargets-release.cmake:90 (set_target_properties):
  set_target_properties Can not find target to add properties to:
  eCAL::mon_plugin_lib
Call Stack (most recent call first):
  C:/eCAL/lib/cmake/eCAL/eCALTargets.cmake:161 (include)
  C:/eCAL/lib/cmake/eCAL/eCALConfig.cmake:62 (include)
  CMakeLists.txt:9 (find_package)
```

**To Reproduce**
1. Download binary release for eCAL and install it on Windows
2. CMake against it

**Desktop (please complete the following information):**
 - OS: Windows
 - Version master @ g5f75e928

**Comment by KerstinKeller** [comment]
@FlorianReimold 

### Please be aware of the implication of these changes.
### Td:lr You cannot use eCAL anymore without having Qt in your system!

## The long answer
Mainly, now we have a strong dependency of ecAL SDK to QT, because `eCAL::mon_plugin_lib` also links QT libraries, if Qt is not available (or there is not `find_package(QT)` call, a `find_package(eCAL)` call will fail.
This is also the case if your not doing anything with `eCAL::mon_plugin_lib`, but just writing a pure eCAL communication example.
We need to avoid this at all cost!

We should strongly consider to use CMake `find_package(eCAL COMPONENT core)` or `find_package(eCAL COMPONENT apps)` components for this.
This means, that all components have to create their own `ecal-core-targets.cmake`, `ecal-apps-targets.cmake` file, and `ecal-targets.cmake` includes these different generated file.

Basically, this is what Boost and other larger libraries do.
This also means that you can also do `find_package(eCAL-core REQUIRED)` which is equivalent to `find_package(eCAL REQUIRED COMPONENT core)`.

Also, and I know that all of you will disagree, I think that we should move away from prebuild sdks, and at the very least split sdk and tools installers.
Instead, rather spend some time on integration with package managers.
